spotdl/search/provider.py

```diff
@@ -95,7 +95,7 @@ def __map_result_to_song_data(result: dict) -> dict:
         'name': result['title'],
         'type': result['resultType'],
         'artist': artists,
-        'length': __parse_duration(result['duration']),
+        'length': __parse_duration(result.get('duration', None)),
```

Comment:
The `None` is a default return value from `get()` function. You don't have to specify it.
```suggestion
        'length': __parse_duration(result.get('duration')),
```